File: pocketbase/pb_migrations/030_create_customer_tracking_collection.js
/// <reference path="../pb_data/types.d.ts" />

/**
 * Migration: Create customer tracking collection
 *
 * Creates customer_tracking collection for tracking customer interactions and status
 */
migrate((app) => {
  // 获取customers集合的ID
  const customersCollection = app.findCollectionByNameOrId("customers");
  if (!customersCollection) {
    throw new Error("Customers collection not found");
  }

  const collection = new Collection({
    name: "customer_tracking",
    type: "base",
    system: false,
    fields: [
      {
        name: "customer_id",
        type: "relation",
        required: true,
        collectionId: customersCollection.id,
        cascadeDelete: false,
        maxSelect: 1
      },
      {
        name: "status",
        type: "select",
        required: false,
        maxSelect: 1,
        values: ["Active", "Lead", "Follow-up", "Onboarded"]
      },
      {
        name: "priority",
        type: "select",
        required: false,
        maxSelect: 1,
        values: ["Low", "Medium", "High"]
      },
      {
        name: "contact_status",
        type: "select",
        required: false,
        maxSelect: 1,
        values: ["Contacted", "Replied", "No Reply"]
      },
      {
        name: "next_action_icon",
        type: "select",
        required: false,
        maxSelect: 1,
        values: ["event", "schedule", "warning", "check_circle", "calendar", "clock", "alert_triangle", "check"]
      },
      {
        name: "next_action_text",
        type: "text",
        required: false,
        max: 200
      },
      {
        name: "next_step_action",
        type: "text",
        required: false,
        max: 100
      },
      {
        name: "next_step_date",
        type: "date",
        required: false
      },
      {
        name: "notes",
        type: "text",
        required: false,
        max: 2000
      },
      // Standard fields
      {
        name: "created_by",
        type: "relation",
        required: false,
        collectionId: "_pb_users_auth_",
        cascadeDelete: false,
        maxSelect: 1
      },
      {
        name: "updated_by",
        type: "relation",
        required: false,
        collectionId: "_pb_users_auth_",
        cascadeDelete: false,
        maxSelect: 1
      }
    ],
    listRule: "@request.auth.id != ''",
    viewRule: "@request.auth.id != ''",
    createRule: "@request.auth.id != ''",
    updateRule: "@request.auth.id != ''", // 允许授权用户更新，而不仅仅是创建者
    deleteRule: "@request.auth.id != ''" // 允许授权用户删除
  });

  return app.save(collection);
}, (app) => {
  const collection = app.findCollectionByNameOrId("customer_tracking");
  return app.delete(collection);
});

File: migrations/030_create_customer_tracking_collection.go
// migrations/030_create_customer_tracking_collection.go
package migrations

import (
	"github.com/pocketbase/dbx"
	"github.com/pocketbase/pocketbase/daos"
	m "github.com/pocketbase/pocketbase/migrations"
	"github.com/pocketbase/pocketbase/models"
	"github.com/pocketbase/pocketbase/tools/types"
)

func init() {
	m.Register(func(db dbx.Builder) error {
		dao := daos.New(db)

		collection, err := dao.FindCollectionByNameOrId("customers")
		if err != nil {
			return err
		}

		// 创建 customer_tracking 集合
		customerTrackingCollection := &models.Collection{}
		customerTrackingCollection.Id = ""
		customerTrackingCollection.Name = "customer_tracking"
		customerTrackingCollection.Type = models.CollectionTypeBase
		customerTrackingCollection.System = false
		customerTrackingCollection.Schema = schema.NewSchema(
			schema.NewTextField("customer_id", "Customer ID", true, false, false),
			schema.NewSelectField("status", "Status", false, false, []string{"Active", "Lead", "Follow-up", "Onboarded"}, "Lead", nil),
			schema.NewSelectField("priority", "Priority", false, false, []string{"Low", "Medium", "High"}, "Medium", nil),
			schema.NewSelectField("contact_status", "Contact Status", false, false, []string{"Contacted", "Replied", "No Reply"}, "Contacted", nil),
			schema.NewSelectField("next_action_icon", "Next Action Icon", false, false, []string{"event", "schedule", "warning", "check_circle", "calendar", "clock", "alert_triangle", "check"}, "calendar", nil),
			schema.NewTextField("next_action_text", "Next Action Text", false, false, false),
			schema.NewTextField("next_step_action", "Next Step Action", false, false, false),
			schema.NewDateField("next_step_date", "Next Step Date", false, false, "", types.DateMax, types.DateMin),
			schema.NewTextField("notes", "Notes", false, false, true),
		)
		customerTrackingCollection.ListRule = types.Pointer("")
		customerTrackingCollection.ViewRule = types.Pointer("")
		customerTrackingCollection.CreateRule = types.Pointer("")
		customerTrackingCollection.UpdateRule = types.Pointer("")
		customerTrackingCollection.DeleteRule = types.Pointer("")
		customerTrackingCollection.Options = types.ModelOptions{
			IntervalBetweenCreations: 0,
			AllowUsernameLogin:       false,
			RequireEmailConfirmation: false,
		}

		return dao.SaveCollection(customerTrackingCollection)
	}, func(db dbx.Builder) error {
		dao := daos.New(db)

		err := dao.DeleteCollection("customer_tracking")
		if err != nil {
			return err
		}

		return nil
	})
}